src/platform_impl/redox/mod.rs

```diff
@@ -0,0 +1,1336 @@
+                                            device_id: event::DeviceId(DeviceId),
+                                        },
+                                    },
+                                    &self.window_target,
+                                    &mut control_flow,
+                                );
+                            }
+                        }
+                        other => {
+                            warn!("unhandled event: {:?}", other);
+                        }
+                    }
+                }
+
+                if count == event_buf.len() {
+                    // If event buf was full, process same window again to ensure all events are drained
+                    continue;
+                }
+
+                // Acknowledge the latest resize event
+                if let Some((w, h)) = resize_opt.take() {
+                    window
+                        .write(format!("S,{},{}", w, h).as_bytes())
+                        .expect("failed to acknowledge resize");
+
+                    // Require redraw after resize
+                    let mut redraws = self.window_target.p.redraws.lock().unwrap();
+                    if !redraws.contains(&window_id) {
+                        redraws.push_back(window_id);
+                    }
+                }
+
+                // Move to next window
+                i += 1;
+            }
+
+            while let Ok(event) = self.window_target.p.user_events_receiver.try_recv() {
+                event_handler(
+                    event::Event::UserEvent(event),
+                    &self.window_target,
+                    &mut control_flow,
+                );
+            }
+
+            event_handler(
+                event::Event::MainEventsCleared,
+                &self.window_target,
+                &mut control_flow,
+            );
+
+            // To avoid deadlocks the redraws lock is not held during event processing
+            while let Some(window_id) = {
+                let mut redraws = self.window_target.p.redraws.lock().unwrap();
+                redraws.pop_front()
+            } {
+                event_handler(
+                    event::Event::RedrawRequested(window::WindowId(window_id)),
+                    &self.window_target,
+                    &mut control_flow,
+                );
+            }
+
+            event_handler(
+                event::Event::RedrawEventsCleared,
+                &self.window_target,
+                &mut control_flow,
+            );
+
+            let mut requested_resume = None;
+            let wait = match control_flow {
+                ControlFlow::Poll => false,
+                ControlFlow::Wait => true,
+                ControlFlow::WaitUntil(instant) => {
+                    requested_resume = Some(instant);
+                    true
+                }
+                ControlFlow::ExitWithCode(code) => break code,
+            };
+
+            if wait {
```

Comment:
If can do `continue` from the match, no need for `wait` variable.

Reply:
Done